internal_packages/keybase/lib/decrypt-button.cjsx

```diff
@@ -29,80 +32,112 @@ class DecryptMessageButton extends React.Component
+      )
+    else
+      @_openPassphrasePopover(popoverTarget, @decryptPopoverDone)
+
+  _displayError: (err) ->
+    dialog = remote.dialog
+    dialog.showErrorBox('Decryption Error', err.toString())
 
   _onClickDecryptAttachments: (event) =>
     popoverTarget = event.target.getBoundingClientRect()
-
-    Actions.openPopover(
-      <PassphrasePopover onPopoverDone={ @_decryptAttachmentsPopoverDone } />,
-      {originRect: popoverTarget, direction: 'down'}
-    )
-
-  _decryptPopoverDone: (passphrase) =>
-    {message} = @props
-    for recipient in message.to
+    if @_noPrivateKeys()
+      Actions.openPopover(
+        <PrivateKeyPopover
+          addresses={_.pluck(@props.message.to, "email")}
+          callback={() => @_openPassphrasePopover(popoverTarget, @decryptAttachmentsPopoverDone)}/>,
+        {originRect: popoverTarget, direction: 'down'}
+      )
+    else
+      @_openPassphrasePopover(popoverTarget, @decryptAttachmentsPopoverDone)
+
+  decryptPopoverDone: (passphrase) =>
+    for recipient in @props.message.to
       # right now, just try to unlock all possible keys
       # (many will fail - TODO?)
       privateKeys = PGPKeyStore.privKeys(address: recipient.email, timed: false)
       for privateKey in privateKeys
         PGPKeyStore.getKeyContents(key: privateKey, passphrase: passphrase)
 
-  _decryptAttachmentsPopoverDone: (passphrase) =>
-    {message} = @props
-    for recipient in message.to
+  decryptAttachmentsPopoverDone: (passphrase) =>
+    for recipient in @props.message.to
       privateKeys = PGPKeyStore.privKeys(address: recipient.email, timed: false)
       for privateKey in privateKeys
         PGPKeyStore.getKeyContents(key: privateKey, passphrase: passphrase, callback: (identity) => PGPKeyStore.decryptAttachments(identity, @state.encryptedAttachments))
 
+  _openPassphrasePopover: (target, callback) =>
+    Actions.openPopover(
+        <PassphrasePopover addresses={_.pluck(@props.message.to, "email")} onPopoverDone={callback} />,
+        {originRect: target, direction: 'down'}
+      )
+
+  _noPrivateKeys: =>
+    numKeys = 0
+    for recipient in @props.message.to
+      numKeys = numKeys + PGPKeyStore.privKeys(address: recipient.email, timed: false).length
+    return numKeys < 1
+
   render: =>
-    # TODO inform user of errors/etc. instead of failing without showing it
     if not (@state.wasEncrypted or @state.encryptedAttachments.length > 0)
       return false
 
-    # TODO a message saying "this was decrypted with the key for ___@___.com"
-    title = if @state.isDecrypted then "Message Decrypted" else "Message Encrypted"
+    title = "Message Encrypted"
+    decryptLabel = "Decrypt"
+    borderClass = "border"
+    decryptClass = "decrypt-bar"
+    if @state.status?
+      console.warn(@state.status)
+      if @state.status.indexOf("Message decrypted") >= 0
+        title = @state.status
+        borderClass = "border done-border"
+        decryptClass = "decrypt-bar done-decrypt-bar"
```

Comment:
It might make sense to boil this all down to a single variable, like `class=done`, and then say "border #{class}-border", "decrypt-bar #{class}-decrypt-bar", etc.